pkg/resources/custom-smtp/custom-smtp.go

```diff
@@ -0,0 +1,216 @@
+package custom_smtp
+
+import (
+	"context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/addon"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	Valid        ValidationResponse = "valid"
+	Partial      ValidationResponse = "partial"
+	Blank        ValidationResponse = "blank"
+	CustomSecret string             = "custom-smtp"
+)
+
+type ValidationResponse string
+
+type CustomSmtp struct {
+	FromAddress string
+	Address     string
+	Password    string
+	Port        string
+	Username    string
+}
+
+func GetCustomAddonValues(serverClient k8sclient.Client, namespace string) (*CustomSmtp, error) {
+
+	addonName := addon.GetName(v1alpha1.InstallationTypeManagedApi)
+	secretName := fmt.Sprintf("addon-%s-parameters", addonName)
+
+	secret := &corev1.Secret{}
+	err := serverClient.Get(context.TODO(), k8sclient.ObjectKey{
+		Name:      secretName,
+		Namespace: namespace,
+	}, secret)
+
+	if err != nil {
+		return nil, err
+	}
+
+	customSmtp := &CustomSmtp{}
+
+	value, ok := secret.Data["custom-smtp-from_address"]
+	if ok {
+		customSmtp.FromAddress = string(value)
+	}
+
+	value, ok = secret.Data["custom-smtp-address"]
+	if ok {
+		customSmtp.Address = string(value)
+	}
+
+	value, ok = secret.Data["custom-smtp-password"]
+	if ok {
+		customSmtp.Password = string(value)
+	}
+
+	value, ok = secret.Data["custom-smtp-port"]
+	if ok {
+		customSmtp.Port = string(value)
+	}
+
+	value, ok = secret.Data["custom-smtp-username"]
+	if ok {
+		customSmtp.Username = string(value)
+	}
+
+	return customSmtp, nil
+}
+
+func ParameterValidation(smtp *CustomSmtp) ValidationResponse {
+
+	valid, partial := false, false
+
```

Comment:
Would be worth checking the validation more in-depth rather than just checking if it's empty or not?
I wonder what's going to happen if I enter some random string in the port value?

Reply:
The reason I ask is that we are saying a config is valid, although from what it looks to me, it might not be valid, it's just that it has all the values set.

Reply:
While yes your points are valid, there are a few blockers to doing that. Currently we don't have a good way to validate the configuration short from sending emails with the configuration. The question about the port being a number is good. We are creating a follow up JIRA to improve our validate. I will make note of this in that ticket.

Reply:
Related Jira https://issues.redhat.com/browse/MGDAPI-3965